dags/public-health/covid19/get_help_to_esri.py

```diff
@@ -0,0 +1,236 @@
+import datetime
+import os
+from urllib.parse import urljoin
+
+import arcgis
+import pandas
+import requests
+from airflow import DAG
+from airflow.hooks.base_hook import BaseHook
+from airflow.models import Variable
+from airflow.operators.python_operator import PythonOperator
+from arcgis.gis import GIS
+
+API_BASE_URL = "https://api2.gethelp.com/v1/"
+
+FACILITIES_ID = "51a351e257374ed3a7776612c7eb0c6a"
+
+STATS_ID = "9db2e26c98134fae9a6f5c154a1e9ac9"
+
+TIMESERIES_ID = "0235713060e74aca95f34ae2b861285f"
+
+
+def upload_to_esri(df, layer_id, filename="/tmp/df.csv"):
+    """
+    A quick helper function to upload a data frame
+    to ESRI as a featurelayer backed CSV
+
+    recommend: no geometries, lat/long columns
+    remember ESRI is UTC only.
+    """
+    df.to_csv(filename, index=False)
+    # Login to ArcGIS
+    arcconnection = BaseHook.get_connection("arcgis")
+    arcuser = arcconnection.login
+    arcpassword = arcconnection.password
+    gis = GIS("http://lahub.maps.arcgis.com", username=arcuser, password=arcpassword)
+
+    gis_item = gis.content.get(layer_id)
+    gis_layer_collection = arcgis.features.FeatureLayerCollection.fromitem(gis_item)
+    gis_layer_collection.manager.overwrite(filename)
+
+    os.remove(filename)
+    return True
+
+
+def make_get_help_request(api_path, token, params={}):
+    """
+    Makes an API request to the GetHelp platform.
+    Also handles depagination of long responses.
+
+    Parameters
+    ==========
+    api_path: string
+        The path to query
+    token: string
+        The OAuth bearer token
+    params: dict
+        Any additional query parameters to pass
+
+    Returns
+    =======
+    The depaginated JSON response in the "content" field.
+    """
+    endpoint = urljoin(API_BASE_URL, api_path)
+    content = []
+    page = 0
+    while True:
+        r = requests.get(
+            endpoint,
+            headers={"Authorization": f"Bearer {token}"},
+            params=dict(page=page, **params),
+        )
+        res = r.json()
+        content = content + res["content"]
+        if res["last"] is True:
+            break
+        else:
+            page = page + 1
+
```

Comment:
assert here to check if data has been pull and API didn't error?

Reply:
actually nvm we would catch that with future asserts

Reply:
Yeah, there are definitely things that could be hardened here, I don't think this is the last time this DAG will need work.